src/nvim/tui/term_input.inl

```diff
@@ -236,19 +235,25 @@ static void read_cb(RStream *rstream, void *rstream_data, bool eof)
     if (handle_bracketed_paste(input) || handle_forced_escape(input)) {
       continue;
     }
-    char *ptr = rbuffer_read_ptr(input->read_buffer);
-    size_t len = rbuffer_pending(input->read_buffer);
+
     // Find the next 'esc' and push everything up to it(excluding)
-    size_t i;
-    for (i = ptr[0] == ESC ? 1 : 0; i < len; i++) {
-      if (ptr[i] == '\x1b') {
+    size_t count;
+    RBUFFER_EACH(buf, c, i) {
```

Comment:
`input->read_buffer` instead of `buf`?

The new code pushes everthing up to the next 'esc' (including).